packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -277,26 +277,25 @@ const makeRenderer = (
 		};
 		const checkReady = state =>
 			state.preRenderChecklist.every(isReady => isReady);
-		const populateStore = store =>
-			new Promise((resolve, reject) => {
+		const populateStore = store => {
+			return addFlags(store).then(() => {
 				// dispatch SERVER_RENDER to kick off API middleware
 				store.dispatch({ type: SERVER_RENDER, payload: url });
 
-				if (checkReady(store.getState())) {
-					addFlags(store).then(() => {
-						resolve(store);
+				return new Promise((resolve, reject) => {
+					// dispatch SERVER_RENDER to kick off API middleware
+					store.dispatch({ type: SERVER_RENDER, payload: url });
```

Comment:
this will be a duplicate call to `store.dispatch` - you've already triggered the action on line 283

Reply:
My lesson today is that no change is ever "simple." :)